core/modules/node/node.pages.inc

```diff
@@ -729,6 +729,11 @@ function node_preview_banner_form(array $form, array $form_state, Node $node, $n
  * @since 1.11.0
  */
 function node_preview_banner_form_node_submit(array $form, array &$form_state) {
+  // Prevent the "This is a preview." and "Changes are stored temporarily."
+  // warnings (previously set in node_preview_banner_form()) from bleeding onto
+  // the saved node page, after saving content changes from the preview mode.
+  backdrop_get_messages();
```

Comment:
I wonder if it's worth editing the comment here to describe *how* this works... Inline comments are for the benefit of developers, but as a developer I'm not sure how or why this line fixes the problem of duplicated messages...

Reply:
@BWPanda if any developer has a quick look at https://api.backdropcms.org/api/backdrop/1/search/backdrop_get_messages they will see that the default values of the two parameters of that function are basically clearing the message queue. I understand that the name of that function may be misleading, but the help for the second parameter does a pretty good job at explaining things:

> **bool $clear_queue**: (optional) If this is TRUE, the queue will be cleared of messages of the type specified in the $type parameter. Otherwise the queue will be left intact. Defaults to TRUE.

...combined with the help for the first parameter, I believe that it's easy to put one and one together:

> **string $type**: (optional) Limit the messages returned by type. Defaults to NULL, meaning all types.

I may be wrong re thins being obvious, and I shouldn't be making assumptions. I have no idea how to explain this though. Care to provide a suggestion please?

Reply:
```suggestion
  // Clear the warning messages displayed in the preview form
  // (node_preview_banner_form()) to prevent confusion after saving.
  backdrop_get_messages('warning', TRUE);
```

Ok, so this change (hopefully) makes it more obvious that we're only clearing the 'warning' messages (setting the 2nd parameter to `TRUE` explicitly helps bring attention to this possibly-overlooked detail).

Reply:
Although I agree with this @BWPanda, this will only work so long as the only type of messages we show in preview mode is warnings. If we happen to add another type on message in the future, then these will bleed onto the save page (which was the whole point of fixing the bug to begin with).

How about using `backdrop_get_messages(NULL, TRUE);` then, which clears all types of messages? That will be more future-proof I believe.

Alternatively, we could do something like this, which explicitly explains what `backdrop_get_messages()` does w/o any parameters passed to it:

```suggestion
  // Use backdrop_get_messages() to clear all messages displayed in the preview
  // form (see node_preview_banner_form()), and prevent them from being
  // displayed in the saved node page.
  backdrop_get_messages();
```

Reply:
I was thinking that to be safer we should just clear 'warning' messages. If we ever change the type of messages above, we can change this too as needed. But I was worried about clearing other types of messages that might be set elsewhere on the site and displayed here (for whatever reason), so figured just clearing warnings was better than clearing everything.

Re. the parameters, I feel that explicitly defining them makes the code easier to read/understand. If `backdrop_get_messages()` was better named for the purpose it's being used for here (e.g. `backdrop_clear_messages()`), then that'd be fine to use as-is. But since it's not obvious how it's working in this context without further investigation, then I think the parameters are useful.

Reply:
Thanks @BWPanda 🙏 ...I've moved the messages to `node_view()` instead + added some logic to only show them when this is a node preview, so all this is not required any more